src/capstone.rs

```diff
@@ -311,7 +327,7 @@ pub struct Details {
 pub enum DetailsArch {
     X86(cs_x86),
     ARM64(cs_arm64),
-    ARM(cs_arm),
+    ARM(Box<cs_arm>),
```

Comment:
I'm curious why this needs to be in a Box? I'll take a look later.

Reply:
So basically: ARM(cs_arm) is the largest enum variant that contains 1768 bytes atleast. The second largest is: X86(cs_x86) which is only 464 bytes. Since enum size is bounded by its largest variant, having one large variant can penalize the memory layout of that enum. So Boxing that largest variant can possible reduce the enum size now to 464 bytes, but will automatically increase based on the Box variant when necessary.

Makes sense?

Btw, I use ARM

Reply:
However if you feel you dont want it to be boxed, we can remove the box, I just tried to save a little memory

Reply:
Yeah, I'm going to remove the box, but let's get this merged first :)!